doc/rst/users-guide/datapar/promotion.rst

```diff
@@ -60,6 +60,50 @@ which would again be equivalent to the loop:
   :language: chapel
   :lines: 25-26
 
+Expressions returning scalar results that are promoted are only evaluated once.
+Consider an alternative ``negate()``, ``negateAndReturn()``, which returns the
+negated ``real``:
+
+.. literalinclude:: examples/users-guide/datapar/promotion.chpl
+  :language: chapel
+  :lines: 199-201
+
+As before, we can call the function in a promoted manner using an array of
+reals, as follows:
+
+.. literalinclude:: examples/users-guide/datapar/promotion.chpl
+  :language: chapel
+  :lines: 203
+
+This is equivalent to:
+
+.. literalinclude:: examples/users-guide/datapar/promotion.chpl
+  :language: chapel
+  :lines: 206-207
+
+Calling this function with a single ``real`` and then assigning to an array of
+reals works similarly:
+
+.. literalinclude:: examples/users-guide/datapar/promotion.chpl
+  :language: chapel
+  :lines: 210
+
+If we print out the array we see that the single real has been negated and then
+duplicated across the array:
+
+.. literalinclude:: examples/users-guide/datapar/promotion.good
+  :language: chapel
+  :lines: 52
+
+The execution is equivalent to:
+
+.. literalinclude:: examples/users-guide/datapar/promotion.chpl
+  :language: chapel
+  :lines: 212-214
+
+Note the ``tmp`` variable, the function ``negateAndReturn`` is only called once
+and then the value is assigned to each element in the array.
+
```

Comment:
This description feels a bit roundabout and potentially confusing to me.  Specifically, it reads (to me) as though `negateAndReturn()` is the thing being potentially promoted in the description in lines 63-65, particularly since it goes on to show a promoted call of it in the lines that follow.  Yet in the statement:

```chapel
A = negateAndReturn(7.8);
```

the thing that's being promoted—if anything (I'll come back to this) is the `=` operator in `A = scalarExpression;`  So saying that `negateAndReturn(7.8)` is only called once in this example feels almost as pointless as saying that it would only be called once for each call in these statements:

```chapel
writeln(negateAndReturn(7.8));
var x = negateAndReturn(7.8) * negateAndReturn(9.0);
var x, y, z = negateAndReturn(10.11);
...
```

I.e., "Why wouldn't it be called once—it's a function call?" (and one that doesn't involve any promotion).  So what's really interesting is the `=` assignment itself, but that leads to my caveat above:

Why did I say "if anything" above?  Because we actually implement `array-of-t = scalar-of-type-t` assignment using an overload of `=` that accepts an array as its LHS and a scalar as its RHS, so it's not actually a promotion.

I think the clearer way to make this point would be to focus on a case where an explicitly declared function is being promoted where it takes two (or more) arguments and doesn't promote one (or more) of them.  That would probably suggest moving it down into the Multi-Argument Promotion section—in fact, it feels like a very natural  place to transition to after the last paragraph which talks about how some arguments may not be involved in the promotion while others are?